### PR TITLE
Steelpoint Donor Armour Update

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

Effects a sprite update to the M4X Armour (Donor Armour). Also changes the item description to be more in-universe, and adds alt-camo options for the armour based on the map type. 

All credit to @Zenith for the sprite work. Camo sprites are my doing. 

Updates to donor gear are allowed based on my review of past PRs. 

# Explain why it's good for the game

Updating 2015 sprites to 2023 standards is good. 

# Testing Photographs and Procedure

Screenshot 2023-12-20 12 43 35

# Changelog
:cl: Zenith, Steelpoint
add: Updates Steelpoint's donor armour, replacing it with a new sprite, camouflage options for all maps and new item descriptions that are lore friendly. 
/:cl:

### CODE REVIEW

**Comment by mullenpaul** [comment]
code lgtm

File: code/modules/cm_marines/Donator_Items.dm [hunk 1]
```diff
@@ -1319,10 +1320,10 @@
 	item_state = "doom_boots"
 
 /obj/item/clothing/shoes/marine/fluff/steelpoint //CKEY=steelpoint (UNIQUE)
-	name = "M4X Boot"
-	desc = "Boots to the M4X.  DONOR ITEM"
-	icon_state = "jackboots"
-	item_state = "jackboots"
+	name = "M4-X Boot"
+	desc = "Standard issue boots issued alongside M4-X armour, features a special coating of acid-resistant layering to allow its operator to move through acid-dretched enviroments safely. This prototype version lacks that feature.  DONOR ITEM"
```
Comment (mullenpaul) [inline code/modules/cm_marines/Donator_Items.dm:1324]: Inconsistency - armour and armor are used interchangeably.

File: code/modules/cm_marines/Donator_Items.dm [hunk 2]
```diff
@@ -1148,8 +1149,8 @@
 	flags_jumpsuit = FALSE
 
 /obj/item/clothing/under/marine/fluff/steelpoint //CKEY=steelpoint (UNIQUE)
-	name = "M4X Jumpsuit"
-	desc = "Jumpsuit to the M4X!!!  DONOR ITEM"
+	name = "M4-X Jumpsuit"
+	desc = "Jumpsuit issued alongside the M4-X armor. Considered outdated compared to the more modern armour system.  DONOR ITEM"
```
Comment (mullenpaul) [inline code/modules/cm_marines/Donator_Items.dm:1153]: Armor and armour both used.

File: code/modules/cm_marines/Donator_Items.dm [hunk 3]
```diff
@@ -842,10 +842,11 @@
 	flags_inv_hide = HIDEEARS|HIDEMASK|HIDEEYES|HIDEALLHAIR
 
 /obj/item/clothing/head/helmet/marine/fluff/steelpoint //CKEY=steelpoint (UNIQUE)
-	name = "M4X Helmet"
-	desc = "Helmets to the M4X!!!  DONOR ITEM"
+	name = "M4-X Helmet"
+	desc = "A next generation combat helmet intended to be paired with the M4-X armour. The full faced helmet provides complete light ballistic-resistant protection alongside enchanced acid resistance. This prototype version lacks those features. DONOR ITEM"
```
Comment (mullenpaul) [inline code/modules/cm_marines/Donator_Items.dm:846]: Armour used, I think we should be abiding by US English - armor.

File: code/modules/cm_marines/Donator_Items.dm (hunk 1 above)
Comment (Steelpoint) [reply code/modules/cm_marines/Donator_Items.dm:1324]: Curses of my Australian language strike again

**Comment by Steelpoint** [comment]
Grammar corrected

**Review by Drulikar — APPROVED** [review]
Code changes look fine.

**Review by nauticall — APPROVED** [review]
fits overall but as advice please be mindful of injecting camouflage bits into armor, we don't do that on normal armor/uniforms for a reason. 👍

**Pull request merged**